src/librustc/mir/interpret/allocation.rs

```diff
@@ -100,7 +102,7 @@ impl<Tag> Allocation<Tag> {
     /// Creates a read-only allocation initialized by the given bytes
     pub fn from_bytes<'a>(slice: impl Into<Cow<'a, [u8]>>, align: Align) -> Self {
         let bytes = slice.into().into_owned();
-        let size = Size::from_bytes(bytes.len() as u64);
+        let size = Size::from_bytes(u64::try_from(bytes.len()).unwrap());
```

Comment:
Would it be reasonable to make `Size::from_bytes` work for any `impl TryInto<u64>`?

Reply:
Sounds plausible, although it looks like we're dying for trait that combines `Into<u64>` and `usize` *sigh*.

Reply:
> although it looks like we're dying for trait that combines Into<u64> and usize sigh.

Yeah, agreed. But that's a sweeping change.